iceoryx_posh/test/moduletests/test_mepoo_shm_safe_unmanaged_chunk.cpp

```diff
@@ -43,14 +43,12 @@ class ShmSafeUnmanagedChunk_test : public Test
     SharedChunk getChunkFromMemoryManager()
     {
         auto chunkSettingsResult = iox::mepoo::ChunkSettings::create(sizeof(bool), alignof(bool));
-        EXPECT_FALSE(chunkSettingsResult.has_error());
-        if (chunkSettingsResult.has_error())
-        {
-            return nullptr;
-        }
+        iox::cxx::Ensures(!chunkSettingsResult.has_error());
```

Comment:
You could think about adding a message to the user with `iox::cxx::Ensures(!chunkSettingsResult.has_error() && "OMG Why did I fail");`. Same for the other occurrences.

Reply:
The only thing I could add is something like `This should never happen!`. It is there because I`m a chicken and didn't just want to use `.value()`. But for such cases I think we should add an `T& expects(cxx::string)` to the `optional` and `expected`. This would have the same result and is nicer to use in tests where this values are under our control and we can actually expect that they don't fail but somehow want to appease the inner chicken.